File: 5tip/desk_mobil/Cw4/app/src/main/java/com/example/cw4/Nwd.kt
package com.example.cw4

class Nwd {
    fun nwdIter(a:Int,b:Int):Int{
        var temp = a;
        var _a = a;
        var _b = b;
        while(_b != 0){
            temp = _a;
            _a = _b;
            _b = temp % _b;
        }
        return _a;
    }
}

File: 2tip/2tip_des/winForm_UTest/WinFormsApp/Models/Nwd.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinFormsApp.Models
{
    public class Nwd
    {
        public static int NwdRec(int a, int b) {
            a = Math.Abs(a);
            b = Math.Abs(b);
            if (b == 0) return a;
            return NwdRec(b, a % b);
        }

        public static int NwdIter(int a, int b) {
            a = Math.Abs(a);
            b = Math.Abs(b);
            if (b == 0) return a;
            while (b != 0) {
                int temp = a;
                a = b;
                b = temp % a;
            }
            return a;
        }
    }
}
